Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ContestsController.cs

```diff
@@ -375,43 +368,79 @@ public ActionResult ChangeActiveParticipantsEndTime(ChangeTimeForParticipantsVie
                 return this.RedirectToAction<ContestsController>(c => c.Index());
             }
 
-            var notUpdatedUsersUsernames = this.participantsBusiness
-                .GetAllParticipantsWhoWouldBeReducedBelowDefaultContestDuration(
-                    model.ContesId,
-                    model.TimeInMinutes,
-                    model.ParticipantsCreatedAfterDateTime.Value,
-                    model.ParticipantsCreatedBeforeDateTime.Value)
-                .Select(u => u.User.UserName)
-                .ToList();
-
-            this.participantsBusiness
-                .UpdateContestEndTimeForAllParticipantsByContestByParticipantContestStartTimeRangeAndTimeIntervalInMinutes(
-                    model.ContesId,
-                    model.TimeInMinutes,
-                    model.ParticipantsCreatedAfterDateTime.Value,
-                    model.ParticipantsCreatedBeforeDateTime.Value);
+            ServiceResult result;
 
-            var minutesForDisplay = model.TimeInMinutes.ToString();
+            if (model.ChangeByTimeInterval)
```

Comment:
Does this have to be a single `Action`? As is it effectively combines two different methods, using a `flag` and `if statement` to differentiate between the required action and execute the appropriate logic with the only benefit I see is to call common error-handling only once. I think it would be cleaner if it was split in two actual methods.